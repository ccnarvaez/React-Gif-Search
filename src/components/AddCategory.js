import React,  { useState } from "react";
import propTypes from 'prop-types';



export const  AddCategory = ( { setCategories } ) => {

    const [ inputValue, setInputValue ] = useState('hello world!');
    const handleInputChange = (e) => {
        setInputValue(e.target.value)

    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if ( inputValue.trim().length >= 1){
            setCategories(cats => [inputValue,...cats ])
            setInputValue ('');
        }
        
    }
    return(
        < form onSubmit = { handleSubmit}>
            
            <input 
                type= "text"
                value = { inputValue }
                onChange = { handleInputChange }
            />
        </form>
    )
};

AddCategory.propTypes = {
    setCategories: propTypes.func.isRequired
}