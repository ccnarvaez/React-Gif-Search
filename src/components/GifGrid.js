import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";


export const GifGrid = ( { category }) => {

    const { data, loading } = useFetchGifs( category );
    
    return(
    <>
        <h1> { category } </h1>

        { loading && <div className = "card animate__animated animate__rubberBand"> Cargando...</div>}
        <div className = " card-grid">
          
                {data.map( (img ) => (
                        
                        <GifGridItem key = { img.id } title = { img.title } url = { img.url } />
                    )
                )}   
         
        </div>
    </>
    )
}