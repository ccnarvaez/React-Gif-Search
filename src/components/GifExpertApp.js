import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';



 export const GifExpertApp = () => {

   
   let [ categories, setCategories] = useState( [ 'Sailor Moon'] );
 
  

    return(
        <div>
            <h1> Buscador de gifs! </h1>
            <AddCategory setCategories = { setCategories }  />
            <hr />
    
            <ol>
                { categories.map(  category => {
                    return(
                        <GifGrid 
                            category = { category } 
                            key = { category }
                        />
                    )
                })}
            </ol>
        </div>
    )
}