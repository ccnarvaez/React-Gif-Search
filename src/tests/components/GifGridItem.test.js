
import { shallow } from "enzyme";
import React from "react";
import { GifGridItem } from "../../components/GifGridItem";




describe( ' Testing <GifGridItem /> component ', () => {

    // General variables
    const title = 'sailor moon GIF';
    const url = 'https://giphy.com/gifs/sailor-moon-r-26gBjmGEsrFQlj8g8';
    const wrapper = shallow(

        <GifGridItem title = { title } url = { url } />
    )

    test( ' Should render properly' , () => {
 
        expect ( wrapper ).toMatchSnapshot()

    });

    test( ' Should print a paragraph containing title text ', () => {

        const findParagraph = wrapper.find('p').text().trim();

        expect ( findParagraph ).toBe( title );


    });

    test( 'Should content img src and alt props', () => {

        // const img = wrapper.find('#test').html(); 
        // console.log ( <img id="test" src="https://giphy.com/gifs/sailor-moon-r-26gBjmGEsrFQlj8g8" alt="sailor moon GIF"/> )

        const img = wrapper.find('#test').props();
        expect( img.src ). toBe( url );
        expect( img.alt ). toBe( title );
    });

    test( 'div className should contain animate__bounce', () => {
        const div = wrapper.find( 'div' ).props();
        const classTxt = 'card animate__animated animate__bounce'
        expect( div.className ). toBe( classTxt );
        expect( div.className.includes('animate__bounce') ). toBe( true );
    })
})