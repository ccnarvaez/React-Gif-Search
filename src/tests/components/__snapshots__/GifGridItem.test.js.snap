// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Testing <GifGridItem /> component   Should render properly 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="sailor moon GIF"
    id="test"
    src="https://giphy.com/gifs/sailor-moon-r-26gBjmGEsrFQlj8g8"
  />
  <p>
     
    sailor moon GIF
     
  </p>
</div>
`;
